pytorch_lightning/trainer/training_io.py

```diff
@@ -281,6 +281,10 @@ def restore(self, checkpoint_path: str, on_gpu: bool):
         if on_gpu:
             model.cuda(self.root_gpu)
 
+        # restore amp scaling
+        if self.use_amp and self.use_native_amp and 'native_amp_scaling_state' in checkpoint:
```

Comment:
@mcarilli sanity check this loading?

Reply:
Looks good if you fix the saving https://github.com/PyTorchLightning/pytorch-lightning/pull/1561/files#r413418705

Like saving, loading should occur either at the very beginning of an iteration (before any training-related `scaler` calls for that iteration) or the very end of an iteration, after `update()`.  It doesn't make a lot of sense to load state dicts at the end of an iteration, but if the saved state originated from a `scaler.state_dict()` call at the end of, say, iteration 1000 ("end of" meaning after iteration 1000's call to `scaler.update()`), then it's ok to call load_state_dict to resume at the beginning of iteration 1001.